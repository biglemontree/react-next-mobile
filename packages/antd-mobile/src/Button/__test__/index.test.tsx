import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { AlipayCircleFill } from 'reactui-mobile-icons'
import { Button } from '..'

describe('Button', () => {
  it('snapshot', () => {
    const component = shallow(
      <Button className="a b" data-y="111">
        button
      </Button>,
    )
    const component2 = shallow(
      <Button capsule icon={AlipayCircleFill}>
        icon
      </Button>,
    )

    expect(toJSON(component)).toMatchSnapshot()
    expect(toJSON(component2)).toMatchSnapshot()
  })

  it('onPress', () => {
    const onPress = jest.fn()
    const component = shallow(<Button onPress={onPress}>button</Button>)

    component.simulate('press')

    expect(onPress).toHaveBeenCalled()
  })

  it('active className', () => {
    const component = shallow(
      <Button className="a" activeClassName="active">
        button
      </Button>,
    )
    expect(toJSON(component)).toMatchSnapshot()

    component.simulate('pressIn')
    expect(component.find('.amd-button').hasClass('a')).toBeTruthy()
    expect(component.find('.amd-button').hasClass('active')).toBeTruthy()
    expect(toJSON(component)).toMatchSnapshot()

    component.simulate('pressOut')
    expect(component.find('.amd-button').hasClass('a')).toBeTruthy()
    expect(component.find('.amd-button').hasClass('active')).toBeFalsy()
    expect(toJSON(component)).toMatchSnapshot()
  })
})
