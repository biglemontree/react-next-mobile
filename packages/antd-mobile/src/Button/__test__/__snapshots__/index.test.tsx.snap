// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button active className 1`] = `
<Touchable
  disabled={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
>
  <div
    aria-disabled={false}
    className="amd-button amd-button-default a"
    role="button"
  >
    <span
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      button
    </span>
  </div>
</Touchable>
`;

exports[`Button active className 2`] = `
<Touchable
  disabled={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
>
  <div
    aria-disabled={false}
    className="amd-button amd-button-active active amd-button-default a"
    role="button"
  >
    <span
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      button
    </span>
  </div>
</Touchable>
`;

exports[`Button active className 3`] = `
<Touchable
  disabled={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
>
  <div
    aria-disabled={false}
    className="amd-button amd-button-default a"
    role="button"
  >
    <span
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      button
    </span>
  </div>
</Touchable>
`;

exports[`Button snapshot 1`] = `
<Touchable
  disabled={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
>
  <div
    aria-disabled={false}
    className="amd-button amd-button-default a b"
    data-y="111"
    role="button"
  >
    <span
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      button
    </span>
  </div>
</Touchable>
`;

exports[`Button snapshot 2`] = `
<Touchable
  disabled={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
>
  <div
    aria-disabled={false}
    className="amd-button amd-button-default amd-button-capsule amd-button-capsule-md amd-button-icon"
    role="button"
  >
    <AlipayCircleFill
      className=""
      size="sm"
    />
    <span
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      icon
    </span>
  </div>
</Touchable>
`;
